frmts/vrt/vrtdataset.cpp

```diff
@@ -1044,6 +1044,121 @@ GDALDataset *VRTDataset::OpenVRTProtocol(const char *pszSpec)
         return nullptr;
     }
 
+    // scan for subdataset/subdataset_n in tokens, close the source dataset and reopen with subdataset if found/valid
+    // make subdataset and subdataset_n mutually exclusive
+    bool bFound_subdataset = false;
+    for (int i = 0; i < aosTokens.size(); i++)
+    {
+        char *pszKey = nullptr;
+        const char *pszValue = CPLParseNameValue(aosTokens[i], &pszKey);
+
+        if (pszKey && pszValue)
+        {
+            char **papszSubdatasets = GDALGetMetadata(poSrcDS, "SUBDATASETS");
+
+            int nSubdatasets = CSLCount(papszSubdatasets);
```

Comment:
querying subdatasets might be "costly" when not needed. So I'd move this within the if (EQUAL(pszKey, "subdataset")) and if (EQUAL(pszKey, "subdataset_n")) cases

Reply:
done